benchmarks/nx-cugraph/pytest-based/bench_algos.py

```diff
@@ -0,0 +1,235 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import time
+
+import networkx as nx
+import pandas as pd
+import pytest
+from cugraph import datasets
+
+# FIXME: promote these to cugraph.datasets
+from cugraph_benchmarking.params import (
+    hollywood,
+    europe_osm,
+    cit_patents,
+    soc_livejournal,
+)
+
+################################################################################
+# Fixtures and helpers
+backend_params = ["cugraph", None]
+
+small_dataset_params = [datasets.karate,
+                        datasets.netscience,
+                        datasets.email_Eu_core,
+                        ]
+medium_dataset_params = [cit_patents,
+                         hollywood,
+                         europe_osm,
+                         ]
+large_dataset_params = [soc_livejournal,
+                        ]
+
+def nx_graph_from_dataset(dataset_obj):
+    """
+    Read the dataset specified by the dataset_obj and create and return a
+    nx.Graph or nx.DiGraph instance based on the dataset is_directed metadata.
+    """
+    create_using = nx.DiGraph if dataset_obj.metadata["is_directed"] else nx.Graph
+    names = dataset_obj.metadata["col_names"]
+    dtypes = dataset_obj.metadata["col_types"]
+    if isinstance(dataset_obj.metadata["header"], int):
+        header = dataset_obj.metadata["header"]
+    else:
+        header = None
+
+    pandas_edgelist = pd.read_csv(dataset_obj.get_path(),
+                                  delimiter=dataset_obj.metadata["delim"],
+                                  names=names,
+                                  dtype=dict(zip(names, dtypes)),
+                                  header=header,
+                                  )
+    G = nx.from_pandas_edgelist(pandas_edgelist,
+                                source=names[0],
+                                target=names[1],
+                                create_using=create_using)
+    return G
+
+
+@pytest.fixture(scope="module", params=small_dataset_params,
+                ids=lambda ds: f"ds={str(ds)}")
```

Comment:
What does `ids=` do here?

Reply:
I added a comment in the latest commit (`ids=` takes a list of items for each param, or a callable, to generate a custom test id for each param instead of using something auto-generated by pytest)